src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -1,15 +1,93 @@
 package org.broadinstitute.dsde.workbench.leonardo.model
-import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
 
-import scala.util.parsing.json.JSONObject
+import java.time.Instant
+import java.util.UUID
 
-case class ClusterRequest(bucketPath: String, serviceAccount: String, labels: Map[String, String])
-case class ClusterResponse(clusterName: String, googleProject: String, clusterId: String, status: String, description: String, operationName: String) // <--- Expand as needed
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
+import org.broadinstitute.dsde.workbench.leonardo.model.ClusterStatus.ClusterStatus
+import org.broadinstitute.dsde.workbench.leonardo.model.ModelTypes.{GoogleBucket, GoogleProject, GoogleServiceAccount}
+import spray.json.{DefaultJsonProtocol, DeserializationException, JsString, JsValue, JsonFormat}
 
-object LeonardoJsonSupport {
-  import spray.json.DefaultJsonProtocol._
+// maybe we want to get fancy later
+object ModelTypes {
+  type GoogleProject = String
+  type GoogleServiceAccount = String
+  type GoogleBucket = String
+}
 
-  implicit val ClusterRequestFormat = jsonFormat3(ClusterRequest)
-  implicit val clusterResponseFormat = jsonFormat6(ClusterResponse)
+object ClusterStatus extends Enumeration {
+  type ClusterStatus = Value
+  val Unknown, Creating = Value
+}
+
+object Cluster {
+  def apply(clusterRequest: ClusterRequest, clusterResponse: ClusterResponse): Cluster = Cluster(
+    clusterId = UUID.fromString(clusterResponse.clusterId),
+    clusterName = clusterResponse.clusterName,
+    googleProject = clusterResponse.googleProject,
+    googleServiceAccount = clusterRequest.serviceAccount,
+    googleBucket = clusterRequest.bucketPath,
+    operationName = clusterResponse.operationName,
+    status = ClusterStatus.Creating,
+    createdDate = Instant.now(),
+    destroyedDate = None,
+    labels = clusterRequest.labels)
+}
+
+case class Cluster(clusterId: UUID,
+                   clusterName: String,
+                   googleProject: GoogleProject,
+                   googleServiceAccount: GoogleServiceAccount,
+                   googleBucket: GoogleBucket,
+                   operationName: String,
+                   status: ClusterStatus,
+                   createdDate: Instant,
```

Comment:
for people reading this: `Instant` is the new `DateTime` now that joda-time has been deprecated in Java 8